src/main/java/codechicken/core/ClassDiscoverer.java

```diff
@@ -114,21 +114,18 @@ private void findClasspathMods() {
     }
 
     private void readFromZipFile(File file) throws IOException {
-        FileInputStream fileinputstream = new FileInputStream(file);
-        ZipInputStream zipinputstream = new ZipInputStream(fileinputstream);
-        do {
-            ZipEntry zipentry = zipinputstream.getNextEntry();
-            if (zipentry == null) {
-                break;
-            }
+        ZipFile zipFile = new ZipFile(file);
+        Enumeration<? extends ZipEntry> zipEntries = zipFile.entries();
+        while (zipEntries.hasMoreElements()) {
+            ZipEntry zipentry = zipEntries.nextElement();
             String fullname = zipentry.getName().replace('\\', '/');
             int pos = fullname.lastIndexOf('/');
             String name = pos == -1 ? fullname : fullname.substring(pos + 1);
             if (!zipentry.isDirectory() && matcher.matches(name)) {
                 checkAddClass(fullname);
             }
-        } while (true);
-        fileinputstream.close();
+        }
+        zipFile.close();
```

Comment:
Can you wrap this block in a try-with-resources block to automatically close the zipFile in an exception-safe way?
```java
try(ZipFile zipFile = new ZipFile(file)) {
  ...
  // calling .close() by hand not needed
}
```